src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -270,6 +285,16 @@ public Opportunity getOpportunity(long id) {
         return opportunity;
     }
 
+    public List<Opportunity> getAllOpportunities() {
+        List<Opportunity> opportunities = new ArrayList<>();
```

Comment:
Good work, you should only get Titles of all opportunites for dashboard insteal of all opportunties for efficiency. Please look at how getMessages function is working.

Reply:
Although I think we are only communicating data in text, it would be easy to render all opportunities at once. However, I will implement the suggested version after some front-end implementation.

Reply:
That would bring us back to the issue (discuss in the team meeting) that if we have 1000+ opportunties, it would be too slow.

Reply:
Em, okay. Then I will retrieve only titles for dashboard and load  full details on-click.

Reply:
Deleted this implementation as per your directions. Re-implemented by modifying the same files of version one.